src/main/java/net/liquidpineapple/pang/screens/LevelEditor.java

```diff
@@ -0,0 +1,227 @@
+      }
+      //key 4 pressed, add ball with size 4
+      if (InputHandler.getKeysDown().contains(52)) {
+        InputHandler.getKeysDown().clear();
+        int size = 4;
+        int offset = 0;
+        try {
+          Image ballImage = ImageIO.read(LevelScreen.class.getResource("/sprites/Balls/blue.png"));
+          offset = ballImage.getWidth(null) / 2;
+        } catch (IOException ioException) {
+          ioException.printStackTrace();
+        }
+        addedObjects.add(new Ball(intXpos - offset, intYpos - offset, ballMovement, size));
+      }
+      //key 5 pressed, add player
+      if (InputHandler.getKeysDown().contains(53) && !addedPlayer) {
+        InputHandler.getKeysDown().clear();
+        addedPlayer = true;
+        int offset = 0;
+        int fixedPlayerYpos = 475;
+        try {
+          Image playerImage = ImageIO.read(LevelScreen.class
+              .getResource("/sprites/player/p1_front.png"));
+          offset = playerImage.getWidth(null) / 2;
+        } catch (IOException ioException) {
+          ioException.printStackTrace();
+        }
+        addedObjects.add(new Player(intXpos - offset, fixedPlayerYpos, new Player1()));
+      }
+      //key p pressed, play the last saved level
+      if (InputHandler.getKeysDown().contains(80)) {
+        InputHandler.getKeysDown().clear();
+        addedPlayer = false;
+        UserCreatedLevels levels = new UserCreatedLevels();
+        Application.getBoard().setLevels(levels.createIterator());
+        Application.getBoard().changeScreen((Screen) Application.getBoard().getLevels().next());
+      }
+      //key s pressed, save the current level
+      if (InputHandler.getKeysDown().contains(83)) {
+        InputHandler.getKeysDown().clear();
+        if (addedPlayer) {
+          createLevel();
+        } else {
+          int fixedPlayerYpos = 475;
+          int fixedPlayerXpos = 380;
+          addedObjects.add(new Player(fixedPlayerXpos, fixedPlayerYpos, new Player1()));
+          addedPlayer = true;
+          createLevel();
+        }
+      }
+    }
+    //select the object
+    GameObject selectedObject = null;
+    if (InputHandler.isLeftMouseButtonDown() || InputHandler.isRightMouseButtonDown()) {
+      for (GameObject addedObject : addedObjects) {
+        if (intXpos > addedObject.getXpos()
+            && intXpos < addedObject.getXpos() + addedObject.getWidth()
+            && intYpos > addedObject.getYpos()
+            && intYpos < addedObject.getYpos() + addedObject.getHeight()) {
+          selectedObject = addedObject;
+        }
+      }
+    }
+
+    //delete an object
+    if (InputHandler.isRightMouseButtonDown() && selectedObject != null) {
+      if (selectedObject instanceof Player) {
+        addedPlayer = false;
+      }
+      deleteObject(selectedObject);
+    }
+
+    //drag and drop
+    while (InputHandler.isLeftMouseButtonDown() && selectedObject != null) {
```

Comment:
~~I just died.~~ Please don't put a while loop like this inside of the doUpdate method. Please.